test/js/ez-asynchronous-block.js

```diff
@@ -56,37 +56,64 @@ describe('ez-asynchronous-block', function() {
     });
 
     describe('load()', function () {
-        beforeEach(function () {
-            sinon.spy(window, 'fetch');
-        });
-
-        afterEach(function () {
-            fetch.restore();
-        });
-
         it('should request the `url`', function () {
+            sinon.spy(window, 'fetch');
             element.load();
             const headers = fetch.firstCall.args[1].headers;
 
             assert.isTrue(fetch.calledOnce);
             assert.isTrue(fetch.alwaysCalledWith(element.url));
-
             assert.equal(
                 'application/partial-update+html',
                 headers.get('Accept'),
                 'The Accept header should be set to get a partial HTML update'
             );
+            fetch.restore();
+
         });
 
         it('should update the content', function (done) {
+            sinon.spy(window, 'fetch');
             element.addEventListener('ez:asynchronousBlock:updated', function () {
                 assert.isNotNull(element.querySelector('.updated'));
                 done();
             });
             element.load();
+            fetch.restore();
+        });
+
+        describe('redirection', function () {
+            beforeEach(function () {
+                const response = new Response(null, {'status': 205});
+
+                sinon.stub(window, 'fetch', function () {
+                    return Promise.resolve(response);
+                });
+            });
+
+            afterEach(function () {
+                fetch.restore();
+            });
+
+            it('should be redirected after 205 status', function (done) {
+                document.addEventListener('ez:navigateTo', function () {
```

Comment:
you need to remove the such *global* event listener otherwise, it will have side effects in following tests.